File: JavaScript/7js 1/divisibleByThree.js
/*
|--------------------------------------------------------------------------
| By 3, or not by 3? That is the question . . .
|--------------------------------------------------------------------------
|
*/

function divisibleByThree(str) {
   return str.split('').map(i => Number(i)).reduce((a, b) => a + b) % 3 == 0;
}


function divisibleByThree(str) {
   return [...str].reduce((s, d) => +d + s, 0) % 3 == 0;
}


function divisibleByThree(str) {
   let s = 0;
   for (let c of str)
      s += parseInt(c);
   return s % 3 == 0;
}


console.log(divisibleByThree('123'), true);
console.log(divisibleByThree('19254'), true);
console.log(divisibleByThree('88'), false);
console.log(divisibleByThree('33333333'), true);

File: Python/7py 1/divisible_by_three.py
"""
|--------------------------------------------------------------------------
| By 3, or not by 3? That is the question . . .
|--------------------------------------------------------------------------
|
"""

def divisible_by_three(st): 
   return sum(list(map(lambda i: int(i), list(st)))) % 3 == 0


def divisible_by_three(st): 
   while len(st) > 1:
      st = str(sum(int(c) for c in st))
   return st in '0369'


def divisible_by_three(st): 
   return divisible_by_three(str(sum(map(int, st)))) if len(st) > 1 else st in '369'


from math import gcd
def divisible_by_three(st): 
   return gcd(sum(map(int, st)), 3) == 3


print(divisible_by_three('123'), True)
print(divisible_by_three('19254'), True)
print(divisible_by_three('88'), False)
print(divisible_by_three('33333333'), True)